src/libsyntax/tokenstream.rs

```diff
@@ -749,67 +884,95 @@ impl TokenSlice {
         Some((path, rst))
     }
 
-    /// Returns an iterator over a TokenSlice (as a sequence of TokenStreams).
-    fn iter(&self) -> Iter {
-        Iter { vs: self }
+
+    /// Compares two TokenStreams, checking equality without regarding span information.
+    pub fn eq_unspanned(&self, other: &TokenStream) -> bool {
+        for (t1, t2) in self.iter().zip(other.iter()) {
+            if !t1.eq_unspanned(t2) {
+                return false;
+            }
+        }
+        true
     }
 
-    /// Splits a TokenSlice based on the provided `&TokenTree -> bool` predicate.
-    fn split<P>(&self, pred: P) -> Split<P>
-        where P: FnMut(&TokenTree) -> bool
-    {
-        Split {
-            vs: self,
-            pred: pred,
-            finished: false,
+    /// Construct a TokenStream from an ast literal.
+    pub fn from_ast_lit_str(lit: ast::Lit) -> Option<TokenStream> {
+        match lit.node {
+            LitKind::Str(val, _) => {
+                let val = TokLit::Str_(token::intern(&val));
+                let vec = vec![TokenTree::Token(lit.span, Token::Literal(val, None))];
+                Some(TokenStream::from_tts(vec))
+            }
+            _ => None,
         }
     }
+
+    /// Convert a vector of TokenTrees into a parentheses-delimited TokenStream.
+    pub fn as_paren_delimited_stream(tts: Vec<TokenTree>) -> TokenStream {
```

Comment:
it should probably take a slice of TTs rather than a Vec. But even so, this seems wrong headed somehow - there should be a ctor-type function for TTs that creates a Delimted one from a slice of TTs and a TS ctor function that takes a single TokenTree and makes a TokenStream. Or something like that.